app/src/main/java/org/mozilla/rocket/urlinput/QuickSearchAdapter.kt

```diff
@@ -0,0 +1,41 @@
+/* -*- Mode: Java; c-basic-offset: 4; tab-width: 20; indent-tabs-mode: nil; -*-
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+package org.mozilla.rocket.urlinput
+
+import android.os.StrictMode
+import android.support.v7.recyclerview.extensions.ListAdapter
+import android.support.v7.widget.RecyclerView
+import android.util.Log
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ImageView
+import kotlinx.android.synthetic.main.quick_search_item.view.quick_search_img
+import org.mozilla.focus.R
+import org.mozilla.icon.FavIconUtils
+
+class QuickSearchAdapter(private val clickListener: (QuickSearch) -> Unit) : ListAdapter<QuickSearch, QuickSearchAdapter.EngineViewHolder>(QuickSearchDiffCallback()) {
+    override fun onCreateViewHolder(parent: ViewGroup, i: Int): EngineViewHolder {
+        val itemView = LayoutInflater.from(parent.context).inflate(R.layout.quick_search_item, parent, false)
+        return EngineViewHolder(itemView)
+    }
+
+    override fun onBindViewHolder(viewHolder: EngineViewHolder, i: Int) {
+        viewHolder.bind(getItem(i), clickListener)
+    }
+
+    class EngineViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
+        var icon: ImageView = itemView.quick_search_img
+
+        fun bind(item: QuickSearch, clickListener: (QuickSearch) -> Unit) {
+            val threadPolicy = StrictMode.allowThreadDiskWrites()
+            val resource = FavIconUtils.getBitmapFromUri(itemView.context, item.icon)
+            icon.setImageBitmap(resource)
+            Log.e("ramoss", "name = " + item.name)
```

Comment:
Wondering what ramoss is 🤔

Reply:
This is debugging log, I forgot to remove. Already add a [PR](https://github.com/mozilla-tw/FirefoxLite/pull/3165) for this.